lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -444,31 +440,15 @@ public Attachment newAttachment() {
         return new AttachmentImpl();
     }
 
-    static void setPrimaryObjRef(TaskImpl task) {
-        ObjectReference objRef = new ObjectReference();
-        objRef.setCompany(task.getPorCompany());
-        objRef.setSystem(task.getPorSystem());
-        objRef.setSystemInstance(task.getPorSystemInstance());
-        objRef.setType(task.getPorType());
-        objRef.setValue(task.getPorValue());
-        task.setPrimaryObjRef(objRef);
-    }
-
-    private List<Attachment> setAttachmentObjRef(List<AttachmentImpl> attachments) {
-        List<Attachment> results = new ArrayList<>();
-        if (attachments != null && !attachments.isEmpty()) {
-            for (AttachmentImpl attachment : attachments) {
-                ObjectReference objRef = new ObjectReference();
-                objRef.setCompany(attachment.getPorCompany());
-                objRef.setSystem(attachment.getPorSystem());
-                objRef.setSystemInstance(attachment.getPorSystemInstance());
-                objRef.setType(attachment.getPorType());
-                objRef.setValue(attachment.getPorValue());
-                attachment.setObjectReference(objRef);
-                results.add(attachment);
+    private void setAttachments(TaskImpl result) {
+        List<AttachmentImpl> attachmentImpls = attachmentMapper.findAttachmentsByTaskId(result.getId());
+        List<Attachment> attachments = new ArrayList<Attachment>();
+        if (attachmentImpls != null && !attachmentImpls.isEmpty()) {
+            for (AttachmentImpl attImpl : attachmentImpls) {
```

Comment:
can be written as 1line lambda. (optinal)

Reply:
As we saw, lambda performance is worse than the simple loop performance